### PR TITLE
delivery materials e2e tests

### PR DESCRIPTION
**Pull request by ioaNikas**
Delivery Materials CRUD operations tests: issue#987

### CODE REVIEW

File: apps/delivery/delivery-e2e/src/support/pages/delivery/DeliveryEditablePage.ts [hunk 1]
```diff
@@ -17,22 +18,190 @@ export default class DeliveryEditablePage extends NavbarPage {
...
+
+  public deleteMaterial(): ConfirmModal {
+    cy.get('[page-id=delivery-material-form] button[test-id=delete]').click();
+    return new ConfirmModal();
+  }
+
+  public selectMaterial(material: Material) {
+    cy.get('[page-id=delivery-material-list] tr')
+      .contains(material.title)
+      .parent()
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/support/pages/delivery/DeliveryEditablePage.ts]: Dajung found a better solution for do that without use ``parent()``:
```
cy.get('[page-id=delivery-material-list] tr').contains('tr', material.title).find('mat-checkbox').click();
```

File: apps/delivery/delivery-e2e/src/support/pages/delivery/DeliveryEditablePage.ts [hunk 2]
```diff
@@ -17,22 +18,190 @@ export default class DeliveryEditablePage extends NavbarPage {
...
+  }
+
+  public saveMaterial() {
+    cy.get('[page-id=delivery-editable] button[test-id=save]').click();
+  }
+
+  public editMaterial(material: Material) {
+    cy.get('[page-id=delivery-material-list] tr')
+      .contains(material.title)
+      .parent()
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/support/pages/delivery/DeliveryEditablePage.ts]: Same that ``selectMaterial()``

File: apps/delivery/delivery-e2e/src/support/pages/movie/MovieListPage.ts [hunk 3]
```diff
@@ -18,12 +16,12 @@ export default class MovieListPage extends NavbarPage {
   }
 
   public clickOnMovieWithNoDeliveries(movieName: string): StarterPickerPage {
-    cy.get('[page-id=movie-list] mat-card-title').contains(movieName).parent().click();
+    cy.get('[page-id=movie-list] mat-card').contains(movieName).parent().parent().parent().parent().click();
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/support/pages/movie/MovieListPage.ts]: Same

File: apps/delivery/delivery-e2e/src/support/pages/movie/MovieListPage.ts [hunk 4]
```diff
@@ -18,12 +16,12 @@ export default class MovieListPage extends NavbarPage {
...
     return new StarterPickerPage();
   }
 
   public clickOnMovieWithDeliveries(movieName: string): DeliveryListPage {
-    cy.get('[page-id=movie-list] mat-card-title').contains(movieName).parent().click();
+    cy.get('[page-id=movie-list] mat-card').contains(movieName).parent().parent().parent().parent().click();
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/support/pages/movie/MovieListPage.ts]: Same

File: apps/delivery/delivery-e2e/src/support/utils/functions.ts [hunk 5]
```diff
@@ -1,3 +1,29 @@
 export const randomString = (): string => `${Math.floor(Math.random() * 10000000)}`;
 
-export const randomEmail = (): string => `cypress${Math.floor(Math.random() * 10000) + 1}@blockframes.com`;
+export const randomEmail = (): string =>
+  `cypress${Math.floor(Math.random() * 10000) + 1}@blockframes.com`;
+
+export const getCurrencySymbol = (currency: string): string => {
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/support/utils/functions.ts]: Do we really need that ?
We can just verify directly ``$, €``, no ?

File: apps/delivery/delivery-e2e/src/integration/CRUD-delivery.spec.ts [hunk 6]
```diff
@@ -82,7 +84,7 @@ beforeEach(() => {
 });
 
 describe('User create a delivery selecting a movie', () => {
-  it('should login, click on the movie card, click on create from scrash, select "Signature of the delivery", and then create a delivery', () => {
+  it.skip('should login, click on the movie card, click on create from scrash, select "Signature of the delivery", and then create a delivery', () => {
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/integration/CRUD-delivery.spec.ts]: Don't forgot to remove ``skip``

File: apps/delivery/delivery-e2e/src/integration/CRUD-delivery.spec.ts [hunk 7]
```diff
@@ -151,3 +153,69 @@ describe('User delete a delivery', () => {
...
+  });
+});
+
+describe('User update some materials fields', () => {
+  it('should login, click on the second movie card, click on the first delivery, then update 3 materials fields and assert that they exist', () => {
+    const p1: MovieListPage = new MovieListPage();
+    const p2: DeliveryListPage = p1.clickOnMovieWithDeliveries(MOVIES_CYTEST[1]);
+    const p3: DeliveryEditablePage = p2.clickFirstDelivery(ORGANIZATION_NAME);
+    let index = 0;
+    MATERIALS.forEach(material => {
```
Comment (Hugo-Hontans) [inline apps/delivery/delivery-e2e/src/integration/CRUD-delivery.spec.ts]: I think you can add the index directly in the ``forEach``

File: libs/ui/src/lib/movie-card/movie-card.component.html [hunk 8]
```diff
@@ -1,4 +1,4 @@
-<mat-card>
+<mat-card test-id="movie-card">
```
Comment (Hugo-Hontans) [inline libs/ui/src/lib/movie-card/movie-card.component.html]: You don't have the test-id injected by ``@Hostebinding`` in ts file ?

File: apps/delivery/delivery-e2e/src/support/utils/functions.ts (hunk 5 above)
Comment (ioaNikas) [reply apps/delivery/delivery-e2e/src/support/utils/functions.ts]: Yes so we can passe material.currency again to check if the symbol is corresponding to the selected currency.

**Pull request merged**